test/strings/basic.jl

```diff
@@ -1420,3 +1420,20 @@ end
         @test transcode(String, transcode(UInt8, transcode(UInt16, str))) == str
     end
 end
+
+if Sys.iswindows()
+    @testset "cwstring" begin
+        # empty string
+        str_0 = ""
+        # string with embedded NUL character
+        str_1 = "Au\000B"
+        # string with terminating NUL character
+        str_2 = "Wordu\000"
+        # "Regular" string with UTF-8 characters of differing byte counts
+        str_3 = "aܣ𒀀"
+        @test Base.cwstring(str_0) == UInt16[0x0000]
+        @test_throws ArgumentError Base.cwstring(str_1)
+        @test Base.cwstring(str_2) == UInt16[0x0057, 0x006f, 0x0072, 0x0064, 0x0000]
```

Comment:
```suggestion
        @test_throws ArgumentError Base.cwstring(str_2)
```
I do not think we should have automatic NUL trimming on `cwstring`.
- There would be a slight performance cost to check for that
- Generally, `cwstring(s)` produces a `Vector{Cwchar_t}` that represents `s` exactly. If we trimmed trailing NUL characters then the returned value would not fully describe the input
- I don't see a case where someone would have a String or AbstractString that is semantically NUL terminated and want to automatically strip that character.